### PR TITLE
Record current directory on switch instead of exit

### PR DESCRIPTION
**Pull request by davdroman**
Implements #1800

### CODE REVIEW

**Comment by jesseduffield** [comment]
Nice one :)

The shift+Q thing is problematic: I do believe some users are making use of that feature (I myself was for a while but haven't found myself using it lately). We can retain that functionality by saving the current directory open first opening the app and if you hit shift+Q to quit then we'll write that back to the file before quitting.

Only tricky thing is to decide is when to capture it and where to store it. We'll want to capture it as soon as possible i.e. before we switch directories. I believe we first call os.Chdir in `pkg/app/app.go` so we'll need to capture the value before then, and then it's a matter of where to put it.

I think we should pass it to the Gui struct so that that struct has it on hand to write it back before quitting when shift+Q is pressed.

I'm aware this is some unfortunate feature creep for your first PR haha but I think it should still be fairly straightforward (if a little hairy). Lemme know your thoughts :)

**Comment by davdroman** [comment]
@jesseduffield that sounds like a sensible approach. It feels like a slightly more challenging implementation but I can give it a shot and report back if I get lost somewhere.

**Comment by jesseduffield** [comment]
Awesome, sounds good

**Comment by davdroman** [comment]
@jesseduffield alright here we go! Now the old functionality is retained along with the new 👍

File: pkg/gui/gui.go [hunk 1]
```diff
@@ -124,6 +124,8 @@ type Gui struct {
 	PopupHandler PopupHandler
 
 	IsNewRepo bool
+
+	InitialRepoDir string
```
Comment (jesseduffield) [inline pkg/gui/gui.go:128]: could we rename this to just `InitialDir`, just because it's not necessarily going to be the root directory of the repo

**Review by jesseduffield — CHANGES_REQUESTED** [review]
Awesome work, just one comment

**Comment by jesseduffield** [comment]
@davdroman I've taken the liberty of updating that field name so that we can get this PR merged :) Reason being that I'm about to merge in the mother of all refactor PRs and I'd hate to yet again force you to update the code

**Comment by davdroman** [comment]
@jesseduffield thank you! I was going to do it this morning but haven't arrived to the office yet. By all means feel free to merge :)

**Pull request merged**